source/initial_composition/world_builder.cc

```diff
@@ -0,0 +1,69 @@
+/*
+  Copyright (C) 2011 - 2017 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#ifdef ASPECT_USE_WORLD_BUILDER
+#include <aspect/initial_composition/world_builder.h>
+#include <aspect/geometry_model/interface.h>
+#include <world_builder/world.h>
+
+
+namespace aspect
+{
+  namespace InitialComposition
+  {
+
+    template <int dim>
+    double
+    WorldBuilder<dim>::
+    initial_composition (const Point<dim> &position, const unsigned int n_comp) const
+    {
+      return this->get_world_builder().composition(Utilities::convert_point_to_array(position),
+                                                   this->get_geometry_model().depth(position),
+                                                   n_comp) ? 1.0 : 0.0;
```

Comment:
I do not understand the final `? 1.0 : 0.0`. Why does the world build not simply return the correct compositional value? E.g. what if you want to have a material that is 0.5  of two compositions at the same time?

Reply:
hmm, I have changed my mind on this several times now. It is through this mechanism possible to have several compositions at the same time, but they all return true or false. I could change it back, so that it returns a double, but I have a hard time coming up with cases where you want to start a model with multiple compositions at the same location. Do you have something in mind?

Reply:
Sorry for the late reply, was travelling. The prime example would be having grain size (or finite strain) in some compositional fields, and actual (chemical) compositions in the other fields. Also porosity in melt models is a separate field that is independent of the other fields.